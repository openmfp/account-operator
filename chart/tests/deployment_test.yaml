suite: operator
templates:
  - deployment.yaml
  - service-account.yaml
values:
  - ../test-values.yaml
tests:
  - it: operator match the snapshot
    asserts:
      - matchSnapshot: {}