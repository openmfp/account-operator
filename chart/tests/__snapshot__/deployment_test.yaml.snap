operator match the snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-cr
    rules:
      - apiGroups:
          - core.openmfp.io
        resources:
          - accounts
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - core.openmfp.io
        resources:
          - accounts
          - accounts/status
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - get
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-crb
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-cr
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-sa
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          service: RELEASE-NAME
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "false"
          labels:
            control-plane: controller-manager
            service: RELEASE-NAME
        spec:
          containers:
            - args:
                - --leader-elect
                - --log-level=
                - --health-probe-bind-address=:8081
              image: :latest
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 9443
                  name: webhook-server
                  protocol: TCP
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 260m
                  memory: 512Mi
                requests:
                  cpu: 150m
                  memory: 128Mi
              securityContext:
                runAsNonRoot: true
          serviceAccountName: RELEASE-NAME-sa
          terminationGracePeriodSeconds: 10
  4: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: RELEASE-NAME-sa
      namespace: NAMESPACE
